### ISSUE TITLE
Draw a single item

### ISSUE THREAD

**Issue by Flaflo** [edayot/SimpleDrawer#3]
At the moment you can only draw full stacks from a drawer, sometimes its much better if you could draw items one by one.

**Comment by edayot** [comment]
Hi,
Unfortunatly, minecraft don't allow datapack to detect attacking on a entity such as item frames and if you think about it I use all possible interaction : 
- Interact clicking with something in hand for filling
- Interact clicking with nothing in hand for unfilling
- Interact clicking will sneaking with something in hand for filling your entire inventory in the drawer
- Interact clicking will sneaking with nothing in hand for unfilling the drawer in all the player inventory

**Comment by Flaflo** [comment]
Would it be a possiblity to change the interacting mechanic at its core?

```
Interact with something in hand -> insert one item
Interact with nothing in hand -> draw one item
Interact + sneak with something in hand -> insert whole stack
Interact + sneak with nothing in hand -> draw whole stack
```

And just drop the "entire inventory" mechanic.

If not we can get a bit creative we could use some sort of toggable mode (How exactly can be discussed):

Mode 1 (Default):
```
Interact with something in hand -> insert whole stack
Interact with nothing in hand -> draw whole stack
Interact + sneak with something in hand -> insert whole inventory
Interact + sneak with nothing in hand -> draw everything into inventory
```

Mode 2:
```
Interact with something in hand -> insert one item inside
Interact with nothing in hand -> draw one item
Interact + sneak with something in hand -> insert whole stack
Interact + sneak with nothing in hand -> draw whole stack
```

Maybe the mode can be toggled by double sneak (sneak fast 2 times while looking at any drawer) and actionbar will display the current mode while looking at any drawer

The mode is global and not per drawer and unique per player

**Comment by edayot** [comment]
I'm taking note of your suggestion, I've already tried to implement putting only one item and also a stack and also a whole inventory in a drawer at the same time, a global mode is maybe a good solution. But there are also other solutions, a long click or a double click to put a stack.

**Comment by Flaflo** [comment]
> I'm taking note of your suggestion, I've already tried to implement putting only one item and also a stack and also a whole inventory in a drawer at the same time, a global mode is maybe a good solution. But there are also other solutions, a long click or a double click to put a stack.

Yea there are several solutions to this, we just need to be creative! :D

**Comment by edayot** [comment]
Implemented in : 356074c5a5d303eb9f597d2ef5d9eedec01f2509 
Now there is a config gui in chat to choose between input/output mode 

(the gui was the last thing to implement all config was done 2 month ago)
